designs/Prerelease-Option.md

```diff
@@ -0,0 +1,69 @@
+# Spec Name
+
+* Status: In Review
+* Author(s): [Martin Ruiz](https://github.com/martinrrm)
+* Issue: [4699](https://github.com/NuGet/Home/issues/4699) dotnet add package - support for NuGet "Pre" version
+
+## Problem Background
+
+Customers want to add the latest version of a package to their project, even if the latest version is prerelease.
+
+## Who are the customers
+
+All dotnet.exe customers
+
+## Goals
+
+Add an option to the command `add package` to enable installing latest prerelease version.
+
+## Non-Goals
+
+## Solution
+
+Add the option `--prerelease` in the command `add package`. This command will get the latest version of the package available even if it is not a stable version.
+
+When searching for the latest version available, only listed packages will be considered.
+
+### Scenarios
+
+Packages available
+
+```
+package 2.0.0
```

Comment:
Is 0.9 considered prerelease?
given versions:
0.8
0.9

with --prerelease -> 0.9
without -> fail ?

Reply:
I don’t think so and @nkolev92 can confirm. The definition of prerelease is a version with “-“ in it.

Reply:
Nope, as @anangaur mentioned, prerelease requires a `-` in the version. 
`0.9` is just a weird stable :D 

For reference: https://nugettoolsdev.azurewebsites.net/5.5.0-preview.2.6382/parse-version?version=0.9

Reply:
Is there any formatting that includes the - for a non-preview version or doesn't have the - for a preview version?  Is that formatting enforced in other places as well and documented?

Reply:
> Is there any formatting that includes the - for a non-preview version or doesn't have the - for a preview version?

Nope, that's per the semver specification. :) 
Documented here: https://docs.microsoft.com/en-us/nuget/concepts/package-versioning

Reply:
Looks like I can restore a package without 1.0 with *. So that should be fine

```
<PackageReference Include="xunit.analyzers" Version="*" />
```